packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -399,21 +399,34 @@ export class DeltaManager
         const oldValue = this.readonly;
         this._forceReadonly = readonly;
         if (oldValue !== this.readonly) {
-            let reconnect = false;
             if (this.readonly === true) {
                 // If we switch to readonly while connected, we should disconnect first
                 // See comment in the "readonly" event handler to deltaManager set up by
                 // the ContainerRuntime constructor
-                reconnect = this.disconnectFromDeltaStream("Force readonly");
+
+                if (this.shouldJoinWrite()) {
+                    // If we have pending changes, then we will never send them - it smells like
+                    // host logic error.
+                    this.logger.sendErrorEvent({ eventName: "ForceReadonlyPendingChanged" });
+                }
+
+                this.disconnectFromDeltaStream("Force readonly");
             }
             safeRaiseEvent(this, this.logger, "readonly", this.readonly);
-            if (reconnect) {
+            if (this.reconnectMode === ReconnectMode.Enabled) {
```

Comment:
Aren't we digressing from the logic here? Previously we used to disconnect and check if we need to reconnect and then triggered connection. Now we are disconnectFromDeltaStream if this.readOnly is true but below we are just reconnecting based on ReconnectMode even if we didn't disconnected first? Is it ok to do so?

Reply:
Yep, it's wrong :) I'll fix it - thanks for catching it!